test/uk/gov/hmrc/nisp/controllers/FullNIRecordControllerSpec.scala

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.nisp.controllers
```

Comment:
This has created a lot of duplicated code which isn't testing anything different, which seems a little unnecessary.
The general rule is one spec per controller.
All that this requires is tests testing that full years show a breakdown when the flag is set. This can be done in the NIRecordControllerSpec in isolation by creating new controllers in the tests.